src/Migrations/M028MovePageHeaderSideBarOptions.php

```diff
@@ -0,0 +1,222 @@
+                                href="' . $page_header_button_link . '"
+                                target=' . ($page_header_button_link_checkbox === 'on' ? "_blank" : "") . '
+                                rel="noreferrer noopener"
+                            >
+                                ' . $page_header_button_title . '
+                            </a>
+                        </div>
+                    <!-- /wp:buttons -->'
+                : '') . '
+
+                </div>
+                <figure class="wp-block-media-text__media">
+                <img
+                    src=' . $page_header_image . '
+                    alt=""
+                    class="wp-image-' . $page_header_image_id . ' size-full"
+                />
+                </figure></div>
+                <!-- /wp:media-text --></div>
+                <!-- /wp:group --></div>
+                <!-- /wp:group -->
+                <!-- /wp:planet4-block-templates/page-header -->
+                ';
+
+                $updated_post_content = $content . $page->post_content;
+
+                $keys_to_update = array(
+                    'p4_title',
+                    'p4_subtitle',
+                    'background_image',
+                    'background_image_id',
+                    'p4_description',
+                    'p4_button_title',
+                    'p4_button_link',
+                    'p4_button_link_checkbox',
+                    'p4_hide_page_title_checkbox',
+                );
+
+                // Update specific meta values to empty strings
+                foreach ($keys_to_update as $meta_key) {
+                    if ($meta_key === 'p4_hide_page_title_checkbox' && !empty($page_header_title)) {
+                        if (isset($page_meta_data[$meta_key]) && in_array('on', $page_meta_data[$meta_key])) {
+                            continue;
+                        }
+
+                        // Update meta value with string 'on'
+                        update_post_meta($page->ID, $meta_key, 'on');
+                    } else {
+                        update_post_meta($page->ID, $meta_key, '');
+                    }
+                }
+
+
+                $post_args = array(
+                    'post_content' => $updated_post_content,
+                    'ID' => $page->ID,
+                );
+
+                wp_update_post($post_args);
+
+                continue;
+            }
+
+            if (empty($page_header_image_id) || empty($page_header_title)) {
+                if (
+                    empty($page_header_image_id)
+                    && empty($page_header_title)
+                    && empty($page_header_description)
+                    && empty($page_header_button_title)
+                ) {
+                    continue;
+                }
+
+                $content = '
+                    <!-- wp:group {"layout":{"type":"flex","orientation":"vertical"}} -->
```

Comment:
missing a `</div>` with l.178 to close the group if there's no page header button link; I think you should open the group div independent of `!empty($page_header_title)`, so 
```
                $content = '
                    <!-- wp:group {"layout":{"type":"flex","orientation":"vertical"}} -->
                    <div class="wp-block-group">
                        ' . (!empty($page_header_title) ?
```
and close it with the closing comment, so it's easier to follow
```
                    </div>
                    <!-- /wp:group -->
                ';
```